controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -1527,10 +1524,59 @@ bool master_thread::handle_intel_slave_join(Socket *sd, ieee1905_1::CmduMessageR
     return true;
 }
 
-bool master_thread::handle_non_intel_slave_join(Socket *sd,
-                                                std::shared_ptr<ieee1905_1::tlvWscM1> tlvwscM1,
-                                                std::string bridge_mac, std::string radio_mac,
-                                                ieee1905_1::CmduMessageTx &cmdu_tx)
+/**
+ * @brief parse the radio basic capabilities TLV and store the operating class
+ * in the database as supported channels.
+ * 
+ * @param radio_mac radio mac address
+ * @param radio_caps radio basic capabilities TLV received from the remote agent
+ * @return true on success
+ * @return false on failure
+ */
+bool master_thread::autoconfig_wsc_parse_radio_caps(
+    std::string radio_mac, std::shared_ptr<wfa_map::tlvApRadioBasicCapabilities> radio_caps)
```

Comment:
In one of my PRs, I passed this stuff by const ref instead of passing the shared_ptr, i.e. dereferencing it in the caller. There was no discussion about that, but maybe this is something we should also agree on: prefer to pass by ref or by pointer (in case the pointer is known to be non-null, obviously).